### PR TITLE
Update tests for ECMA262 #2216

### PR DESCRIPTION
**Pull request by devsnek**
https://github.com/tc39/ecma262/pull/2216

### CODE REVIEW

**Comment by rwaldron** [comment]
I'll take a look at this and the related PR tomorrow—thanks!

**Review by dstiers11 — DISMISSED** [review]

File: test/built-ins/Function/internals/Call/class-ctor-realm.js [hunk 1]
```diff
@@ -3,17 +3,20 @@
...
-     TypeError exception.
+  Error when invoking a default class constructor, honoring the Realm
+  that the class was defined in.
 features: [cross-realm, class]
 ---*/
 
-var C = $262.createRealm().global.eval('0, class {}');
+const realm = $262.createRealm();
+const C = realm.global.eval('(class {})');
+const TE = realm.global.eval('TypeError');
```
Comment (rwaldron) [inline test/built-ins/Function/internals/Call/class-ctor-realm.js:13]: Can you walk me through this? I'm not sure I understand how the TypeError thrown in this Realm would come from another Realm (thanks in advance)

Comment (devsnek) [reply test/built-ins/Function/internals/Call/class-ctor-realm.js:13]: What TypeError is thrown in the main realm?

Comment (rwaldron) [reply test/built-ins/Function/internals/Call/class-ctor-realm.js:13]: The try/catch below is in the main Realm, so the object it creates will be from the main Realm.

```js
function otherRealm() {
  let frame = document.createElement('iframe');
  document.body.appendChild(frame);
  frame.contentDocument.write('<body>');
  const global = frame.contentWindow;
  return {
    global
  };
}


const other = otherRealm();
const C = other.global.eval('(class {})');

try {
  C();
} catch (error) {
  console.log(error instanceof globalThis.TypeError); // true
  console.log(error instanceof other.global.TypeError); // false
}
```

Or in V8, if you prefer: 

```js
let realmId = Realm.createAllowCrossRealmAccess();
let other = Realm.global(realmId);
const C = other.eval('(class {})');

try {
  C();
} catch (error) {
  print(error instanceof globalThis.TypeError); // true
  print(error instanceof other.TypeError); // false
}
```

Comment (devsnek) [reply test/built-ins/Function/internals/Call/class-ctor-realm.js:13]: right those engines implement the spec as it is currently written, with my pr the error will come from the other realm.

Comment (rwaldron) [reply test/built-ins/Function/internals/Call/class-ctor-realm.js:13]: That's what I missing. Thanks.

**Comment by rwaldron** [comment]
@devsnek now that this has consensus, is it ready to merge?

**Comment by devsnek** [comment]
It should be.

File: test/built-ins/Function/internals/Call/class-ctor-realm.js [hunk 2]
```diff
@@ -3,17 +3,20 @@
...
-assert.throws(TypeError, function() {
+let error;
+try {
   C();
-});
+} catch (e) {
+  error = e;
+}
+
+assert(error instanceof TE);
```
Comment (ExE-Boss) [inline test/built-ins/Function/internals/Call/class-ctor-realm.js:22]: Shouldn’t this use the following?
```suggestion
assert.throws(TE, function() {
  C();
});
```

**Comment by ljharb** [comment]
@rwaldron can this land? test262 tests are intended to land before the spec PR.

**Comment by rwaldron** [comment]
@ljharb There's an unaddressed question from @ExE-Boss

**Review by rwaldron — APPROVED** [review]

**Comment by rwaldron** [comment]
Thanks everyone!

**Pull request merged**